server/events/working_dir.go

```diff
@@ -147,27 +147,43 @@ func (w *FileWorkspace) forceClone(log *logging.SimpleLogger,
 		cloneURL = w.TestingOverrideCloneURL
 	}
 
-	gitVersion, err := GetGitVersion()
-	if err != nil {
-		return "", errors.Wrap(err, "git version")
+	var cmd *exec.Cmd
+
+	// Clone the repo
+	log.Info("cloning repo %q", cloneURL)
+	cmd = exec.Command("git", "clone", cloneURL, cloneDir) // #nosec
+	cmd.Dir = cloneDir
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return "", errors.Wrapf(err, "cloning repo %s: %s", headRepo.SanitizedCloneURL, string(output))
 	}
 
-	cloneCmd, err := CreateGitCloneCommand(gitVersion, cloneURL, cloneDir)
-	if err != nil {
-		return "", err
+	// Checkout the branch for this PR
+	log.Info("checking out branch %q", p.Branch)
+	cmd = exec.Command("git", "checkout", p.Branch) // #nosec
+	cmd.Dir = cloneDir
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return "", errors.Wrapf(err, "checking out branch %q: %s", p.Branch, string(output))
 	}
 
-	if output, err := cloneCmd.CombinedOutput(); err != nil {
-		return "", errors.Wrapf(err, "cloning %s: %s", headRepo.SanitizedCloneURL, string(output))
+	// Init submodules for the branch
+	// https://blog.github.com/2016-02-01-working-with-submodules
+	// https://git-scm.com/book/en/v2/Git-Tools-Submodules
+	// https://stackoverflow.com/questions/15124430/how-to-checkout-old-git-commit-including-all-submodules-recursively/29156667#29156667
+	log.Info("initializing submodules for branch %q", p.Branch)
+	cmd = exec.Command("git", "submodule", "init") // #nosec
+	cmd.Dir = cloneDir
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return "", errors.Wrapf(err, "initializing submodules for branch %q: %s", p.Branch, string(output))
 	}
 
-	// Check out the branch for this PR.
-	log.Info("checking out branch %q", p.Branch)
-	checkoutCmd := exec.Command("git", "checkout", p.Branch) // #nosec
-	checkoutCmd.Dir = cloneDir
-	if err := checkoutCmd.Run(); err != nil {
-		return "", errors.Wrapf(err, "checking out branch %s", p.Branch)
+	// Recursively update submodules for the branch
+	log.Info("updating submodules for branch %q", p.Branch)
```

Comment:
Will this work if:
a) no submodules defined
b) without calling `git submoule init` first?

Reply:
a) tested on a repo w/o submodules, the command just returns w/o any errors
b) merged into one command `git submodule update --init --recursive`, tested on repos with and without submodules